Model/Refund/Refund.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Amwal\Payments\Model\Refund;
+
+use Amwal\Payments\Model\Checkout\AmwalCheckoutAction;
+use Amwal\Payments\Model\Config;
+use Amwal\Payments\Model\ErrorReporter;
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\RequestOptions;
+use Magento\Framework\Webapi\Rest\Request;
+use Magento\Sales\Api\OrderRepositoryInterface;
+use Amwal\Payments\Model\AmwalClientFactory;
+use Magento\Framework\Serialize\Serializer\Json;
+use Psr\Log\LoggerInterface;
+
+class Refund extends AmwalCheckoutAction
+{
+
+    private RefundHandler $refundHandler;
+
+    private Request $restRequest;
+
+    private AmwalClientFactory $amwalClientFactory;
+
+    private OrderRepositoryInterface $orderRepository;
+
+    private Json $jsonSerializer;
+
+    /**
+     * @param AmwalClientFactory $amwalClientFactory
+     * @param ErrorReporter $errorReporter
+     * @param Config $config
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        AmwalClientFactory       $amwalClientFactory,
+        ErrorReporter            $errorReporter,
+        Config                   $config,
+        LoggerInterface          $logger,
+        RefundHandler            $refundHandler,
+        Request                  $restRequest,
+        OrderRepositoryInterface $orderRepository,
+        Json                     $jsonSerializer
+    )
+    {
+        parent::__construct($errorReporter, $config, $logger);
+        $this->refundHandler = $refundHandler;
+        $this->restRequest = $restRequest;
+        $this->orderRepository = $orderRepository;
+        $this->amwalClientFactory = $amwalClientFactory;
+        $this->jsonSerializer = $jsonSerializer;
+    }
+
+    /**
+     * Executes the refund request.
+     *
+     * @return bool Whether the refund was successful.
+     */
+    public function execute()
+    {
+        $requestBody = $this->restRequest->getBodyParams();
+        $orderId = $requestBody['order_id'];
+        $refundReason = $requestBody['refund_reason'];
+        $refundAmount = (float)$requestBody['refund_amount'];
+        $refundShippingAmount = (float)$requestBody['refund_shipping_amount'];
+        $refundItems = $requestBody['refund_items'];
+        $order = $this->orderRepository->get($orderId);
+        $requestBody = [
+            'refund_amount' => $refundAmount,
+            'metadata' => ['reason' => !empty($refundReason) ? $refundReason : 'Refund request from Magento by Amwal Payments'],
```

Comment:
Should we make this into a translatable string?
```suggestion
            'metadata' => ['reason' => !empty($refundReason) ? $refundReason : __('Refund request from Magento by Amwal Payments')],
```